torch/nn/qat/modules/conv.py

```diff
@@ -21,13 +22,16 @@ class Conv2d(nn.Conv2d):
 
     def __init__(self, in_channels, out_channels, kernel_size, stride=1,
                  padding=0, dilation=1, groups=1,
-                 bias=True, padding_mode='zeros', qconfig=None):
+                 bias=True, padding_mode='zeros', qconfig=None,
+                 device='cpu', dtype=torch.get_default_dtype()) -> None:
+        factory_kwargs = {'device': device, 'dtype': dtype}
         super().__init__(in_channels, out_channels, kernel_size,
                          stride=stride, padding=padding, dilation=dilation,
-                         groups=groups, bias=bias, padding_mode=padding_mode)
+                         groups=groups, bias=bias, padding_mode=padding_mode,
+                         **factory_kwargs)
         assert qconfig, 'qconfig must be provided for QAT module'
         self.qconfig = qconfig
-        self.weight_fake_quant = qconfig.weight()
+        self.weight_fake_quant = qconfig.weight(factory_kwargs=factory_kwargs)
```

Comment:
Representative sample 2: qat modules require an Observer (passed in as part of the `qconfig`). `factory_kwargs` are passed to the Observer during construction (see sample 3 for more info).